jupyterhub/spawner.py

```diff
@@ -917,8 +920,11 @@ def _signal(self, sig):
     def stop(self, now=False):
         """Stop the single-user server process for the current user.
 
-        If `now` is set to True, do not wait for the process to die.
-        Otherwise, it'll wait.
+        If `now` is False (default), shutdown the server as gracefully as possible,
+        e.g. starting with SIGINT, then SIGTERM, then SIGKILL.
+        If `now` is True, terminate the server immediately.
+
+        The coroutine should return when the process is no longer running.
```

Comment:
IMO we should get rid of all references to 'process' from the base abstract authenticators, since the thing being spawned might not be a process (could be a container, or VM, etc). Unrelated to this PR tho maybe.